src/mmv/mod.rs

```diff
@@ -0,0 +1,583 @@
+    value: u64,
+    string_offset: Option<u64>,
+    metric_offset: Option<u64>,
+    instance_offset: Option<u64>
+}
+
+impl ValueBlk {
+    pub fn value(&self) -> u64 { self.value }
+    pub fn string_offset(&self) -> &Option<u64> { &self.string_offset }
+    pub fn metric_offset(&self) -> &Option<u64> { &self.metric_offset }
+    pub fn instance_offset(&self) -> &Option<u64> { &self.instance_offset }
+}
+
+impl MMVReader for ValueBlk {
+    fn from_reader<R: ReadBytesExt>(r: &mut R) -> Result<Self, MMVDumpError> {
+        let value = r.read_u64::<Endian>()?;
+        let string_offset = r.read_u64::<Endian>()?;
+        let metric_offset = r.read_u64::<Endian>()?;
+        let instance_offset = r.read_u64::<Endian>()?;
+
+        Ok(ValueBlk {
+            value: value,
+            string_offset: {
+                if is_valid_blk_offset(string_offset) { Some(string_offset) }
+                else { None }
+            },
+            metric_offset: {
+                if is_valid_blk_offset(metric_offset) { Some(metric_offset) }
+                else { None }
+            },
+            instance_offset: {
+                if is_valid_blk_offset(instance_offset) { Some(instance_offset) }
+                else { None }
+            },
+        })
+    }
+}
+
+/// Indom block structure
+pub struct IndomBlk {
+    indom: Option<u32>,
+    instances: u32,
+    instances_offset: Option<u64>,
+    short_help_offset: Option<u64>,
+    long_help_offset: Option<u64>
+}
+
+impl IndomBlk {
+    pub fn indom(&self) -> &Option<u32> { &self.indom }
+    pub fn instances(&self) -> u32 { self.instances }
+    pub fn instances_offset(&self) -> &Option<u64> { &self.instances_offset }
+    pub fn short_help_offset(&self) -> &Option<u64> { &self.short_help_offset }
+    pub fn long_help_offset(&self) -> &Option<u64> { &self.long_help_offset }
+}
+
+impl MMVReader for IndomBlk {
+    fn from_reader<R: ReadBytesExt>(r: &mut R) -> Result<Self, MMVDumpError> {
+        let indom = r.read_u32::<Endian>()?;
+        let instances = r.read_u32::<Endian>()?;
+        let instances_offset = r.read_u64::<Endian>()?;
+        let short_help_offset = r.read_u64::<Endian>()?;
+        let long_help_offset = r.read_u64::<Endian>()?;
+
+        Ok(IndomBlk {
+            indom: {
+                if is_valid_indom(indom) { Some(indom) }
+                else { None }
+            },
+            instances: instances,
+            instances_offset: {
+                if is_valid_blk_offset(instances_offset) { Some(instances_offset) }
+                else { None }
+            },
+            short_help_offset: {
+                if is_valid_blk_offset(short_help_offset) { Some(short_help_offset) }
+                else { None }
+            },
+            long_help_offset: {
+                if is_valid_blk_offset(long_help_offset) { Some(long_help_offset) }
+                else { None }
+            }
+        })
+    }
+}
+
+/// Instance block structure
```

Comment:
This part is generally most familiar to people familiar with PMAPI, so please also add corresponding references to the `pmapi.h` file with line numbers